controllers/users.js

```diff
@@ -1,92 +1,116 @@
 /* eslint-disable import/no-unresolved */
-const { ObjectId } = require('mongodb');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
 const User = require('../models/user');
+const { messages } = require('../tools/messages');
 
-//отобразить всех пользователей
+const checkUserAndSend = (user, req, res) => {
+  if (!user) {
+    return res.status(404).send({ message: `${messages.user.id.isNotFound}: ${req.params.id}` });
+  }
+  return res.send({ data: user });
+};
+
+// отобразить всех пользователей
 const getUsers = (req, res) => {
   User.find({})
-      .then(users => res.send({ data: users }))
-      .catch(err => res.status(500).send({ message: err.message }));
+    .then((users) => res.send({ data: users }))
+    .catch((err) => res.status(500).send({ message: err.message }));
 };
 
-//найти пользователя по id
+// найти пользователя по id
 const getUserById = (req, res) => {
-  if (ObjectId.isValid(req.params.id)) {
-    User.findById(req.params.id)
-    .then(user => {
-      if (!user) {
-        return res.status(404).send({ message: `Пользователя с id: ${req.params.id} не существует`})
-      }
-      res.send({ data: user }); })
-    .catch(err => res.status(500).send({ message: `Произошла ошибка при запросе пользователя с id: ${req.params.id}`, error: err.message }));
-  } else {
-    res.status(400).send({ message: 'Неправильный формат id пользователя' });
-  }
+  User.findById(req.params.id)
+    .then((user) => checkUserAndSend(user, req, res))
+    .catch((err) => res.status(500).send({
+      message: `${messages.user.id.isNotFound}: ${req.params.id}`,
+      error: err.message,
+    }));
```

Comment:
Можно лучше: использовать `orFail`.
С этим были вопросы, отвечу здесь. Суть хэлпера в том, что он бросает исключение. Любое исключение ловится `.catch()` блоком, соответственно чтобы не словить ошибку по кодам ответов, нам нужно чтобы в ошибке был указан HTTP код.
Для этого можно создать отдельный модуль, в котором написать новую ошибку. Для этой ситуации нужно 404, поэтому получаем:
```js
// errors/notFoundError.js
class NotFoundError extends Error {
  constructor(message) {
    super(message);
    this.statusCode = 404;
  }
}

module.exports = NotFoundError;
```
Мы указали код ошибки, теперь можем спокойно упрощать текущий код:
```suggestion
    .orFail(() => new NotFoundError(`${messages.user.id.isNotFound}: ${req.params.id}`))
    .then((user) => res.send({ data: user }))
    .catch((err) => res.status(err.statusCode || 500).send({ message: err.message }));
```

Подход с созданием модуля ошибки пригодится в следующей практике. Также его можно использовать и в других контроллерах.

Reply:
> В целом стало лучше, но есть еще ошибки. В тех, местах где были проблемы, я постарался детально разъяснить. Также написал про улучшения.
> 
> **Нужно исправить:**
> 
> 1. Ошибки линтинга.
> 2. При попытке создать пользователя с зарегистрированным `email` возвращается 201.
> 3. Ошибка при авторизации.
> 4. При попытке обновить профиль или аватар возвращается 401.
> 5. Роуты не соответствуют заданию.
> 
> **Можно лучше:**
> 
> 1. Использовать `orFail`.

Всё исправил. Завтра попробую переделать остальные методы в контроллерах с созданием модуля ошибки.